cf2/Asset/Register.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+
+namespace calderawp\calderaforms\cf2\Asset;
+
+
+class Register
+{
+    /**
+     *
+     * @since 1.9.0
+     *
+     * @var string
+     */
+
+    protected $handle;
+    /**
+     *
+     * @since 1.9.0
+     *
+     * @var array
+     */
+    protected $data;
+
+    /**
+     *
+     * @since 1.9.0
+     *
+     * @var string
+     */
+    protected $scriptUrl;
+
+    /**
+     *
+     * @since 1.9.0
+     *
+     * @var string
+     */
+    protected $assetsFilePath;
+
+    /**
+     *
+     *
+     * @since 1.9.0
+     *
+     * @var bool
+     */
+    protected $registered;
+
+    /**
+     * Register constructor.
+     *
+     * @param $handle
+     * @param $coreUrl
+     * @param $corePath
+     * @param array $data
+     *
+     * @since 1.9.0
+     *
+     */
+    public function __construct($handle, $coreUrl, $corePath, array $data = [])
+    {
+        $this->handle = $handle;
+        $this->data = $data;
+        $this->registered = false;
+        $this->setScriptUrl($coreUrl . 'clients/' . $this->handle . '/build/index.min.js');
+        $this->setAssetsFilePath($corePath . '/clients/' . $this->handle . '/build/index.min.asset.json');
+    }
+
+
+    /**
+     *
+     * @return string
+     * @since 1.9.0
+     *
+     */
+    public function getScriptUrl()
+    {
+        return $this->scriptUrl;
+    }
+
+    /**
+     *
+     * @return string
+     * @since 1.9.0
+     *
+     */
+    public function getAssetFilePath()
+    {
+        return $this->assetsFilePath;
+    }
+
+    /**
+     * @return $this
+     * @since 1.9.0
+     *
+     */
+    public function setScriptUrl($scriptUrl)
+    {
+        $this->scriptUrl = $scriptUrl;
+        return $this;
+    }
+
+    /**
+     * @return $this
+     * @since 1.9.0
+     *
+     */
+    public function setAssetsFilePath($assetsFilePath)
+    {
+        $this->assetsFilePath = $assetsFilePath;
+        return $this;
+    }
+
+    /**
+     *
+     * @return array
+     * @since 1.9.0
+     *
+     */
+    protected function getLocalizeData()
+    {
+        return $this->data ? $this->data : [];
+    }
+
+    /**
+     * @return $this
+     * @since 1.9.0
+     *
+     */
+    public function register()
+    {
+
+        $assetFile = file_get_contents($this->getAssetFilePath());
```

Comment:
Should we check if file_exists() for $this->getAssetFilePath() ?

Reply:
Yes.  It should always exist -- if assetts were built. Without it we can't register dependencies. So I will skip regesitration if it doesn't exist, which should be fixed by building dependencies.

Reply:
See: 06fa04278ced2aff3ce77b0f2fc81d77c82f01fa